cnd/src/swap_protocols/metadata_store.rs

```diff
@@ -85,49 +82,87 @@ impl Metadata {
     }
 }
 
-#[derive(Debug, Fail)]
+#[derive(Debug)]
 pub enum Error {
-    #[fail(display = "Metadata already exists")]
-    DuplicateKey,
+    Path(String),
+    Init(String),
+    Connect(String),
+    Load(String),
+    Insert(String),
+    Parse(String),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            // FIXME: is this idiomatic [CoBloX] Rust?
+            Error::Path(msg) => write!(f, "Datastore path error: {}", msg),
+            Error::Init(msg) => write!(f, "Failed to initialize datastore : {}", msg),
+            Error::Connect(msg) => write!(f, "Failed to connect to datastore: {}", msg),
+            Error::Load(msg) => write!(f, "Failed to load record: {}", msg),
+            Error::Insert(msg) => write!(f, "Failed to insert new record: {}", msg),
+            Error::Parse(msg) => write!(f, "Failed to parse stored record: {}", msg),
+        }
+    }
+}
+
+// FIXME: Do we need this?
```

Comment:
Can you compile without it?

Reply:
I think most of your error enums don't implement `std::error:Error`. I don't have a source now but I think this trait is deemed useless by the community.